optaplanner-persistence/optaplanner-persistence-jaxb/src/test/java/org/optaplanner/persistence/jaxb/api/score/json/AbstractScoreJaxbJsonAdapterTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.persistence.jaxb.api.score.json;
+
+import java.io.Serializable;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+import org.eclipse.persistence.jaxb.MarshallerProperties;
+import org.eclipse.persistence.jaxb.UnmarshallerProperties;
+import org.optaplanner.core.api.score.Score;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public abstract class AbstractScoreJaxbJsonAdapterTest {
+
+    // ************************************************************************
+    // Helper methods
+    // ************************************************************************
+
+    protected <S extends Score, W extends TestScoreWrapper<S>> void assertSerializeAndDeserialize(S expectedScore, W input) {
+        System.setProperty("javax.xml.bind.context.factory", "org.eclipse.persistence.jaxb.JAXBContextFactory");
```

Comment:
This is dangerous, especially because it doesn't unset it.
The order in which the unit tests are run can break other unit tests that do XML checking.

Reply:
Later it sets
jaxbMarshaller.setProperty(MarshallerProperties.MEDIA_TYPE, "application/json");

So is that system property still needed?

Reply:
Yes, unfortunately the system property is still needed to use MOXy as the JAXB provider according to the [MOXy docs](https://www.eclipse.org/eclipselink/documentation/2.7/concepts/blocks003.htm#CHDBGGJD). But I see what you're saying about affecting other unit tests when a specific order is chosen.

A quick fix would be to set the system property in the `@Before` method of the MOXy test classes. Maybe there's a better alternative.